c10/core/TensorImpl.h

```diff
@@ -359,12 +367,18 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
     return tid == HIPTensorId() || tid == SparseHIPTensorId();
   }
 
+  bool is_mkldnn() const {
+    return type_id() == MkldnnCPUTensorId();
+  }
+
   int64_t get_device() const {
     // NB: This method is not virtual and tries to avoid dispatches in the common case for perf.
     const auto tid = type_id();
     if (tid == CUDATensorId() || tid == HIPTensorId() || tid == MSNPUTensorId() || tid == XLATensorId()) {
       // TODO: #12934 investigate caching device on TensorImpl to avoid this vdispatch.
       return storage().device().index();
+    } else if (tid == MkldnnCPUTensorId()) {
```

Comment:
Hmm why do we need to add a case for mkldnn here? I mean dense cpu tensor doesn't reach here, mkldnn tensor should be similar.

Reply:
@bddppq It is just for completeness. If the code is not reachable, would delete it.